pyi.py

```diff
@@ -227,6 +231,43 @@ def visit_Index(self, node: ast.Index) -> ast.expr:
             return node.value
 
 
+def _is_list_of_str_nodes(seq: list[ast.expr | None]) -> TypeGuard[list[ast.Str]]:
+    return all(isinstance(item, ast.Str) for item in seq)
+
+
+def _is_bad_TypedDict(node: ast.Call) -> bool:
+    """Evaluate whether an assignment-based TypedDict should be rewritten using class syntax.
+
+    Return `False` if the TypedDict appears as though it may be invalidly defined;
+    type-checkers will raise an error in that eventuality.
+    """
+
+    args = node.args
+    if len(args) < 2:
```

Comment:
It's unlikely that anyone will ever add more than 2 args to a TypedDict, but we should probably let type checkers complain if they do:

```suggestion
    if len(args) != 2:
```

Reply:
Yes, this makes sense, because `total` is keyword-only. But I should probably add test cases with `total` specified as well.